### ISSUE TITLE
Концептуальная схема данных

### ISSUE THREAD

**Issue by M0O0V** [M0O0V/Zhirnov-Coursework#8]
### Сущности:

**Пользователи:** Фамилия, Имя, Отчество.

**Аккаунты:** Номер телефона, Ник, Пароль.

**Отзывы:** Текст отзыва, оценка.

**Мероприятия:** Название, Картинка, Описание, Ссылка на сайт организатора, Адрес, Цена, Продолжительность.

**Типы мероприятия:** Название Типа

**Администраторы:** Ник, Пароль.

**Пользователи** заводят **Аккаунты** (один аккаунт - один пользователь), с них могут оставлять **Отзывы** на **Мероприятия**.
**Отзывы** на **Мероприятия** (ноль или много отзывов на каждое мероприятие). 
**Мероприятия** делятся на **Типы**.
**Администратор** может удалять **Отзывы** и добавлять и удалять **Мероприятия**.

![КС](https://github.com/user-attachments/assets/685c8665-4e95-491e-b7d1-11211350e321)

Мне кажется, что **Администраторы** точно не так должны располагаться и связываться. И нужна ли связь **Аккаунты** - **Мероприятия**, если есть **Аккаунты** - **Отзывы** - **Мероприятия**?

**Comment by wldmr-key** [comment]
Жутковато, да. Разделение пользователя и аккаунта достаточно тяжело обосновать. Различие между Администраторами и аккаунтами - тоже

Связь между аккаунтами и мероприятиями из описания предметной области не следует. Проверь напрвление всех стрелок. Не все правильные

**Comment by M0O0V** [comment]
### Сущности:

**Пользователи:** 
- Ник (уникальный)
- Пароль
- Номер телефона
- Email (опционально)
- Дата регистрации
- Роль (пользователь или администратор)

**Отзывы:** 
- Текст отзыва
- Рейтинг (от 1 до 10)
- Дата создания

**Мероприятия:**
- Название
- Описание
- Дата проведения
- Время начала
- Продолжительность
- Стоимость
- Рейтинг
- Адрес
- Ссылка на сайт организатора
- Изображение
- Статус (активное или неактивное)

**Типы мероприятия:**
- Название типа (кино, театры, концерты)

**Избранные мероприятия:**
- Дата добавления

**Площадка:**
- Название
- Адрес
- Описание
- Телефон представителя
- Email представителя


**Пользователи** заводят регестрируются, и могут оставлять **Отзывы** на **Мероприятия**.
Так же, **Пользователи** могут добавлять мероприятия в **Избранные Мероприятия**
**Отзывы** на **Мероприятия** (ноль или много отзывов на каждое мероприятие). 
Каждое **Мероприятие** имеет одну **Площадку** проведения.
На каждой **Площадке** может быть несколько **Мероприятий**.
**Мероприятия** делятся на **Типы** (1 тип у каждого).
К одному **Типу** может относиться много **Мероприятий**.
**Администратор** может удалять **Отзывы** и добавлять и удалять **Мероприятия**.
**Администратор** это **Пользователь** с ролью **Администратор**


![КС2](https://github.com/user-attachments/assets/2270854c-774c-4f61-b3d3-338c3a9e59c4)

**Comment by wldmr-key** [comment]
значительно лучше. Но стрелки! Стрелки!!!! @Drozdova-Tatiana - помоги человеку!!!

**Comment by Drozdova-Tatiana** [comment]
Я думаю, какая-то проблема со сущностью "избранные мероприятия" и со связью от "избранных мероприятий" к "мероприятиям". У избранного мероприятия есть много мероприятий? 

И возможно атрибут "статус" у мероприятия следует вынести как отдельную сущность подобно "типы мероприятий"?

**Comment by Drozdova-Tatiana** [comment]
Также у одного мероприятия может быть много отзывов, наверное следует перевернуть стрелку "мероприятия" - "отзывы".

**Comment by M0O0V** [comment]
> Также у одного мероприятия может быть много отзывов, наверное следует перевернуть стрелку "мероприятия" - "отзывы".

Согласен, я это и имел ввиду. Как раз тут стрелочка у меня и не правильная. Спасибо

**Comment by M0O0V** [comment]
> Я думаю, какая-то проблема со сущностью "избранные мероприятия" и со связью от "избранных мероприятий" к "мероприятиям". У избранного мероприятия есть много мероприятий?

Каждый пользователь может добавить себе много избранных мероприятий, а каждое мероприятие может быть избранным у многих людей. И надо получается развернуть стрелку от Избранных к Мероприятиям.


> И возможно атрибут "статус" у мероприятия следует вынести как отдельную сущность подобно "типы мероприятий"?

Не, не думаю что статус стоит отдельно выносить. По этому критерию мероприятия ообще выводиться будут. Если они уже закончились, то смотреть их и билеты покупать нельзя.

**Comment by wldmr-key** [comment]
`И надо получается развернуть стрелку от Избранных к Мероприятиям` - когда же ты все перерисуешь, чтобы мы поняли, что ты все понял?

`По этому критерию мероприятия ообще выводиться будут.` - то есть, люди, получается, зря отзывы писали?

**Comment by M0O0V** [comment]
![КС3](https://github.com/user-attachments/assets/95e01475-20c3-4b33-91b2-b989e30053de)

**Comment by wldmr-key** [comment]
ок

но , рейтинг, разве, не автоматически считается?

**Comment by M0O0V** [comment]
Автоматически. Тогда не нужен такой атрибут?

**Comment by wldmr-key** [comment]
не нужен
Администратор у сайта один?
Что представляет из себя модерация отзывов?
